### DISCUSSION TITLE
Reinitialize initial form values

### DISCUSSION THREAD

**Discussion by kwameopareasiedu** [mantinedev/mantine#3326] [Feature requests]
Here's an sample of an update form which requires an object from an API to set initial values

```tsx
import React from "react";
import { useQuery } from "@tanstack/react-query";
import { Stack } from "@mantine/core";
import { useForm } from "@mantine/form";

function UpdateForm() {
  const person = useQuery(
    ["person", "758029845"], 
    () => fetch("/person/758029845").then(res => res.json()),
  );

  const form = useForm({
    initialValues: {
      name: person.data?.name || "",
      age: person.data?.age || "",
      address: person.data?.address || "",
    }
  });

  return (
    <Stack>
      {/* Form elements */}
    </Stack>
  );
}
```

From this example, we can see that the initial values depends on the value of `person`. However, `person` is not initially defined and needs to be fetched from the an API. When the API resolves, I noticed that the initial values of `useForm` does not update.

I'd like to request a feature such that `useForm` can take a list of dependencies (like React's `useEffect`). The internals of `useForm` (such as `initialValues`) should update when the dependencies update.

Something along the lines of this:

```tsx
const form = useForm({
  initialValues: {
    name: person.data?.name || "",
    age: person.data?.age || "",
    address: person.data?.address || "",
  }
}, [person]);
```

PS: I'm aware I can use `form.setFieldValue` in a `useEffect` to achieve the same thing, however if the form fields are a lot that can be quite inconvenient.

**Comment by rtivital** [comment]
The variable is called **initial**Value – it contains values that you've set **initially**. There is no reason to update these values.
If you need to set values to form use `form.setValues`.

**Reply by lukepighetti** [reply]
@rtivital if we do this, the form is marked as dirty. And for some reason, we cannot call `form.setValues` and then `form.resetDirty`. The call to reset dirty appears to be no-op when called directly after set values

**Reply by Elvincth** [reply]
> @rtivital if we do this, the form is marked as dirty. And for some reason, we cannot call `form.setValues` and then `form.resetDirty`. The call to reset dirty appears to be no-op when called directly after set values

same issue here. seems  form.reset(); not able to set initialized as false also.

**Comment by rtivital** [comment]
`form.setInitialValues` handler is available – https://mantine.dev/form/values/#setinitialvalues-handler
To update initialValues in `form.resetDirty`, call it with these values: `form.resetDirty(newInitialValues)`.
